smart-contracts/tasks/deploy/relay-bridge-factory.ts

```diff
@@ -2,30 +2,41 @@ import { task } from 'hardhat/config'
 
 import RelayBridgeFactoryModule from '../../ignition/modules/RelayBridgeFactoryModule'
 import networks from '@relay-protocol/networks'
+import { deployContract } from '../../lib/zksync'
 
 task('deploy:bridge-factory', 'Deploy a relay bridge factory').setAction(
-  async (_params, { ethers, ignition }) => {
+  async (_params, hre) => {
+    const { ethers, ignition, network } = hre
+
     // get args value
     const { chainId } = await ethers.provider.getNetwork()
     const { hyperlaneMailbox, name: networkName } = networks[chainId.toString()]
     console.log(`deploying on ${networkName} (${chainId})...`)
 
-    // deploy the pool using ignition
-    const parameters = {
-      RelayBridgeFactory: {
-        hyperlaneMailbox,
-      },
-    }
-
-    const { relayBridgeFactory } = await ignition.deploy(
-      RelayBridgeFactoryModule,
-      {
-        parameters,
-        deploymentId: `RelayBridgeFactory-${chainId.toString()}`,
+    let relayBridgeAddress: string
+    if (network.zksync) {
+      ;({ address: relayBridgeAddress } = await deployContract(
+        hre,
+        'RelayBridgeFactory',
+        [hyperlaneMailbox]
+      ))
```

Comment:
Does this mean zksync does not work with ignition?

Reply:
yes, to deploy to zksync L2 we have to use this helper. On L1 we use ignition

Reply:
Meh.